src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -1161,6 +1166,59 @@ void InterpreterSystemQuery::waitLoadingParts()
     }
 }
 
+
+void InterpreterSystemQuery::loadPrimaryKeys()
+{
+    if (!table_id.empty())
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY, table_id.database_name, table_id.table_name);
+        StoragePtr table = DatabaseCatalog::instance().getTable(table_id, getContext());
+
+        if (auto * merge_tree = dynamic_cast<MergeTreeData *>(table.get()))
+        {
+            LOG_TRACE(log, "Loading primary keys for table {}", table_id.getFullTableName());
+            try
+            {
+                merge_tree->loadPrimaryKeys();
+            }
+            catch (const Exception & ex)
```

Comment:
(kindly see my other comment)

Reply:
Sorry I am confused abt which scope of changes is mentioned here? the loadPrimaryKeys that you highlighted above now has no logic relevant to concurrent loading...?

Reply:
I think he means this one about the catch statement: https://github.com/ClickHouse/ClickHouse/pull/67733#discussion_r1760772371

Reply:
Looking at this again, I think it would be better to not swallow + log (l. 1186) exceptions at all. Just let them bubble up and inform the user that something went wrong. So let's remove l. 1180 + 1184?